redis/runtime/src/main/java/io/quarkiverse/langchain4j/redis/RedisEmbeddingStore.java

```diff
@@ -0,0 +1,246 @@
+                .map(ignored -> randomUUID())
+                .collect(toList());
+        addAllInternal(ids, embeddings, embedded);
+        return ids;
+    }
+
+    private void addInternal(String id, Embedding embedding, TextSegment embedded) {
+        addAllInternal(singletonList(id), singletonList(embedding), embedded == null ? null : singletonList(embedded));
+    }
+
+    private void addAllInternal(List<String> ids, List<Embedding> embeddings, List<TextSegment> embedded) {
+        if (ids.isEmpty() || ids.size() != embeddings.size() || (embedded != null && embedded.size() != embeddings.size())) {
+            throw new IllegalArgumentException("ids, embeddings and embedded must be non-empty and of the same size");
+        }
+        ds.withTransaction(ds -> {
+            ReactiveTransactionalJsonCommands<String> json = ds.json();
+            int size = ids.size();
+            Uni[] unis = new Uni[size];
+            for (int i = 0; i < size; i++) {
+                String id = ids.get(i);
+                Embedding embedding = embeddings.get(i);
+                TextSegment textSegment = embedded == null ? null : embedded.get(i);
+                Map<String, Object> fields = new HashMap<>();
+                fields.put(schema.getVectorFieldName(), embedding.vector());
+                if (textSegment != null) {
+                    fields.put(schema.getScalarFieldName(), textSegment.text());
+                    fields.putAll(textSegment.metadata().asMap());
+                }
+                String key = schema.getPrefix() + id;
+                unis[i] = json.jsonSet(key, "$", fields);
+            }
+            return Uni.join().all(unis).andFailFast().replaceWithVoid();
+        }).await().indefinitely();
+    }
+
+    @Override
+    public List<EmbeddingMatch<TextSegment>> findRelevant(Embedding referenceEmbedding, int maxResults,
+            double minScore) {
+        String queryTemplate = "*=>[ KNN %d @%s $BLOB AS %s ]";
+        String query = format(queryTemplate, maxResults, schema.getVectorFieldName(), SCORE_FIELD_NAME);
+        // TODO: rewrite to the data source api, but we need a new
+        // method QueryArgs.param(String, byte[]) to get it working
+
+        //        QueryArgs args = new QueryArgs()
+        //                .sortByAscending(SCORE_FIELD_NAME)
+        //                .param("DIALECT", "2")
+        //                .param("BLOB", toByteArray(referenceEmbedding.vector()));
+        //        Uni<SearchQueryResponse> search = ds.search()
+        //                .ftSearch(schema.getIndexName(), query, args);
+        //        SearchQueryResponse response = search.await().indefinitely();
+        Request request = Request.cmd(Command.FT_SEARCH)
+                .arg(schema.getIndexName())
+                .arg(query)
+                .arg("PARAMS")
+                .arg("2")
+                .arg("BLOB")
+                .arg(toByteArray(referenceEmbedding.vector()))
```

Comment:
So, that's what we should do.

What about modifying the param method from the data source to allow byte[] or even float[] ?

Reply:
Yeah, I'm going to, thus the TODO, I just want to get something that works out the door quickly